src/java/org/apache/cassandra/db/rows/UnfilteredRowIteratorWithLowerBound.java

```diff
@@ -245,21 +252,32 @@ private ClusteringBound<?> getPartitionIndexLowerBound()
      */
     private boolean canUseMetadataLowerBound()
     {
-        // Side-note: pre-2.1 sstable stat file had clustering value arrays whose size may not match the comparator size
-        // and that would break getMetadataLowerBound. We don't support upgrade from 2.0 to 3.0 directly however so it's
-        // not a true concern. Besides, !sstable.mayHaveTombstones already ensure this is a 3.0 sstable anyway.
-        return !sstable.mayHaveTombstones() && !sstable.metadata().isCompactTable();
+        if (sstable.metadata().isCompactTable())
+            return false;
+
+        Slices requestedSlices = slices;
+
+        if (requestedSlices.isEmpty())
+            return true;
+
+        if (!isReverseOrder())
```

Comment:
We need a comment here that if the requested slice falls within the covered slice, there's no need to delay producing data.